src/docfx/template/TemplateEngine.cs

```diff
@@ -33,7 +34,13 @@ private TemplateEngine(string templateDir)
             _global = LoadGlobalTokens();
             _schemas = LoadSchemas(schemaDir, _contentTemplateDir);
             _liquid = new LiquidTemplate(_contentTemplateDir);
-            _js = new JavascriptEngine(_contentTemplateDir, _global);
+
+            // TODO: remove JINT after Microsoft.CharkraCore NuGet package
+            // supports linux and macOS: https://github.com/microsoft/ChakraCore/issues/2578
+            _js = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? (IJavaScriptEngine)new ChakraCoreJsEngine(_contentTemplateDir, _global)
```

Comment:
why cast to `IJavaScriptEngine` here? _js is already `IJavaScriptEngine`

Reply:
C# is not smart enough to deduce the type here